src/arrayMethodJoin.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let complete = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        complete += separator !== undefined ? separator : ',';
+      }
```

Comment:
Better add default value to parameter `[].__proto__.join2 = function(separator = ',') {`